### PR TITLE
Fix #1263: bad reference for push-federation-images in e2e-test

### LINKED ISSUE #1263: bad reference to push-federation-images in e2e test
https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md#federation-e2e-tests references build/push-federation-images.sh

I believe this should be federation/develop/push-federation-images.sh

### PR DESCRIPTION
**Pull request by yuexiao-wang**
Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

Fix #1263

1. According to https://github.com/kubernetes/kubernetes/pull/42140,  "build/push-federation-images.sh" shoud be "federation/develop/push-federation-images.sh"

2. " cluster/log-dump.sh" should be "federation/cluster/log-dump.sh"

### CODE REVIEW

File: contributors/devel/e2e-tests.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # End-to-End Testing in Kubernetes
 
-Updated: 5/3/2016
+Updated: 25/10/2017
```
Comment (cblecker) [inline contributors/devel/e2e-tests.md]: Can you just remove this date line? It's not needed when we have git history.

Comment (yuexiao-wang) [reply contributors/devel/e2e-tests.md]: Yes,  good catch.

**Comment by yuexiao-wang** [comment]
@cblecker The PR is update. PTAL

**Comment by yuexiao-wang** [comment]
@madhusudancs PTAL

**Comment by yuexiao-wang** [comment]
@madhusudancs Could you review this PR?

**Comment by yuexiao-wang** [comment]
@madhusudancs @Phillels PTAL

**Pull request merged**